### DISCUSSION TITLE
On Tree Fails on defaultBranchRef

### DISCUSSION THREAD

**Discussion by yakov116** [community/community#24741] [Apps, API and Webhooks]
<pre><code class="lang-auto">{
  repository(owner: "sindresorhus", name: "refined-github") {
    defaultBranchRef {
      target {
        ... on Tree {
          entries {
            name
          }
        }
      }
    }
  }
}
</code></pre>
<p>Returns {}</p>
<p>on Commit works fine</p>

**Comment by nickserv** [comment, marked as answer]
<p>The target GitObject doesn’t implement Tree, you need to get its tree and query the tree’s entries.</p>
<pre><code class="lang-auto">{
  repository(owner: "sindresorhus", name: "refined-github") {
    defaultBranchRef {
      target {
        ... on Commit {
          tree {
            entries {
              name
            }
          }
        }
      }
    }
  }
}
</code></pre>
